### PR TITLE
 feat: create ContractDuration settings

### PR DESCRIPTION
**Pull request by btrd**
#303

### CODE REVIEW

File: config/locales/fr.yml [hunk 1]
```diff
@@ -791,6 +792,22 @@ fr:
...
+            one: "1 durée de contrat"
+            other: "%{count} durées de contrats"
+        new:
+          title: "Ajouter une nouvelle durée de contrat"
+        edit:
+          title: "Modifier la durée de contrat '%{name}'"
+        create:
+          success: "Durée de contrat créée !"
+        update:
+          success: "Durée de contrat mis à jour !"
```
Comment (yannski) [inline config/locales/fr.yml:808]:
```suggestion
          success: "Durée de contrat mise à jour !"
```

File: lib/tasks/maintenance.rake [hunk 2]
```diff
@@ -225,6 +225,40 @@ namespace :maintenance do
...
+
+    # Create missing contract duration
+    (1..11).each do |i|
+      ContractDuration.find_or_create_by!(name: "#{i} mois")
+      ContractDuration.find_or_create_by!(name: "#{i} mois renouvelable")
+    end
+    ContractDuration.find_or_create_by!(name: "1 an")
+    ContractDuration.find_or_create_by!(name: "1 an renouvelable")
+    (2..5).each do |i|
+      ContractDuration.find_or_create_by!(name: "#{i} années")
```
Comment (yannski) [inline lib/tasks/maintenance.rake:242]: make it simple here: replace "année" by "an". Therefore the loop should from 1 to 5.

Comment (btrd) [reply lib/tasks/maintenance.rake:242]: `an` also take an `s`, no ?

File: lib/tasks/maintenance.rake [hunk 3]
```diff
@@ -225,6 +225,40 @@ namespace :maintenance do
...
+      ContractDuration.find_or_create_by!(name: "#{i} années renouvelable")
+    end
+
+    # Assign contract duration to job offer when transition from duration_contract has not been done
+    JobOffer.where.not(duration_contract: nil).map do |job_offer|
+      if job_offer.contract_type.duration
```
Comment (yannski) [inline lib/tasks/maintenance.rake:248]: can you add support of different cases ("an" vs "AN", "ans" vers "ANS") and variant ("années" vs "an")?

File: lib/tasks/maintenance.rake (hunk 2 above)
Comment (btrd) [reply lib/tasks/maintenance.rake:242]: But it should be `an/ans` or `année/années`

Comment (yannski) [reply lib/tasks/maintenance.rake:242]: yep. Or use `pluralize` :)

Comment (yannski) [reply lib/tasks/maintenance.rake:242]: `pluralize` with the `count` option.

**Pull request merged**